app/Http/Controllers/Auth/LoginController.php

```diff
@@ -134,28 +150,30 @@ public function showSecondarySignin()
 
     public function verifySecondarySignin(Request $request)
     {
-        $user = Auth::guard('partial_web')->user();
-
-        if (! $user->hasPassword()) {
-            return $this->authDone($user);
-        }
+        $passwordFieldName = 'password';
 
         $this->validate($request, [
-            'password' => 'required|string',
+            $passwordFieldName => 'required|string',
         ]);
 
-        if (! Auth::attempt(['id' => Auth::guard('partial_web')->user()->id, 'password' => $request->input('password')])) {
+        if (! Auth::attempt(['id' => $this->partialWebUser->id, 'password' => $request->input($passwordFieldName)])) {
             $error = \Illuminate\Validation\ValidationException::withMessages([
-               'password' => ['The supplied password did not match our records'],
+                $passwordFieldName => ['The supplied password did not match our records'],
             ]);
             throw $error;
         }
 
-        return $this->authDone($user);
+        return $this->authDone($this->partialWebUser);
     }
 
     public function authDone(User $user)
     {
-        return redirect()->intended('/');
+        Auth::guard(self::FULL_GUARD)->loginUsingId($user->getKey(), true);
+
+        if ($user->needsToUpdatePassword()) {
+            return redirect()->route('login.set_password');
```

Comment:
Should this route be renamed to `login.password.set` inline with other routes?

Reply:
Yep, done